src/dist/replication/lib/replica_stub.cpp

```diff
@@ -1231,7 +1257,36 @@ void replica_stub::close()
         _log->close();
         _log = nullptr;
     }
+}
 
+std::string replica_stub::get_replica_dir(const char* app_type, global_partition_id gpid) const
```

Comment:
Non-deterministic directory assignment is not good as you have to handle complications such as conflictions & duplications. Instead, I suggest we can simply do some deterministic mapping from gpid to its targeted directory to avoid these problems.